### ISSUE TITLE
Need ability to set up commands to run in response to hooks in finit.conf

### ISSUE THREAD

**Issue by villains** [troglobit/finit#18]
This is just an idea that I would find useful.

For an embedded system I am working, I need to run a script that checks and attaches a UBIFS volume before "/bin/mount -na" is called.
In /etc/fstab, I created an entry that will mount the UBIFS volume and if the volume is not attached, the mount will fail.

To accomplish this, I currently have two options:
1. Patch finit.c to call a hard coded script, or run some hard coded commands, before "/bin/mount -na" (or before plugins are called for HOOK_ROOTFS_UP)
2. Create a plugin to call my script in response to HOOK_ROOTFS_UP

I am probably going to use option 2 for this project.

However in the future, it would be a lot easier if you could do simple things like that right from finit.conf.
For example:

```
hooktask [ROOTFS_UP] /sbin/attachubi.sh   # run in parallel with other hook commands
hookrun [NETWORK_UP] /sbin/dosomething.sh    # run in sequence with other hook commands
```

This feature could probably be implemented as an internal plugin. It would need tighter integration in the conf file parsing code, unlike normal plugins. There could be options in the configure script as well, to enable/disable this feature.

Finally... thanks again for all the work you do on finit!
All the best.

**Comment by villains** [comment]
For now it ended up being easier and faster to patch finit.c by inserting the following code:
```
	/* Run our little UBIFS setup script */
	mode = fmode("/sbin/attachubi.sh");
	if (S_ISEXEC(mode)) {
		run_interactive("/sbin/attachubi.sh", "Attach persistent UBIFS volume");
	}
```
right before this point
```
	_d("Root FS up, calling hooks ...");
	plugin_run_hooks(HOOK_ROOTFS_UP);
```

**Comment by troglobit** [comment]
We have something similar at work, we use JFFS2 for a parttion with config files, certificates, etc. that is mounted at boot. If mount fails we have an external plugin that kicks in at `HOOK_MOUNT_ERROR`, erases that flash partition and reboots.  (Yes, it's treated as a catastrophic failure, but better erase flash than create an RMA)

In your case I would recommend putting your code in a plugin, which can be built out-of-tree, and install that in the Finit plugin dir, then it will load and run automatically on whichever hook point you select.

**Comment by troglobit** [comment]
I do like your idea of being able to add scripts at hook points, that was very clever! :)

**Comment by troglobit** [comment]
I just realized this could be implemented using the new condition support:

```conf
task <hook/rootfs/up> /sbin/attachubi.sh      # run in parallel with other tasks/services
run <hook/network/up> /sbin/dosomething.sh    # run in sequence with other commands
```

**Comment by villains** [comment]
Cool! That looks like exactly what I need. Thank you for the heads up.

Sorry for the late reply. Recently I've been working on some non-Linux projects, and now finally getting back into it. I see you've been very busy as well. Great to see all the new things in finit!

**Comment by troglobit** [comment]
Fixed as of 9e31bd8.  Please note that some hook conditions can only trigger run/tasks listed in /etc/finit.conf, rather than /etc/finit.d/*.conf.
